src/future/mod.rs

```diff
@@ -887,6 +888,25 @@ pub trait Future {
     {
         shared::new(self)
     }
+
+    /// Execute this future "in the background" on the provided spawn handled.
+    ///
+    /// This function will consume this future and pass it to the `spawn` value
+    /// provided. This typically means that this future will be driven to
+    /// completion in the background, and once completed the future will be
+    /// deallocated.
+    ///
+    /// # Panics
+    ///
+    /// This function will panic if the instance of `Executor` provided is
+    /// unable to execute the future `f` provided.
+    fn background<E>(self, executor: &E)
+        where E: Executor<Self>,
+              Self: Future<Item = (), Error = ()> + Sized,
+    {
+        executor.execute(self).expect("failed to execute future in background")
```

Comment:
I agree that we should avoid panicking. An executor being unable to execute a task is a very plausible situation to get into.

The two options are for `background` to return a `Result` OR for it to silently drop the error.

I'm leaning towards just dropping the error because:

a) `background` is a convenience API
b) There is no guarantee that an executor returns an error on execution failure. It's completely acceptable for the executor to silently drop the task as well.